authentication/lib/main.dart

```diff
@@ -0,0 +1,20 @@
+import 'package:authentication/firebase_options.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:firebase_core/firebase_core.dart';
+import 'package:flutter/material.dart';
+
+import 'src/app.dart';
+
+void main() async {
+  WidgetsFlutterBinding.ensureInitialized();
+  await Firebase.initializeApp(
+    options: DefaultFirebaseOptions.currentPlatform,
+  );
+
+  // Both of the following lines are good for testing,
+  // but can be removed for release builds
+  await FirebaseAuth.instance.useAuthEmulator('localhost', 9099);
```

Comment:
This could be an opportunity to use multiple entry point files with IDE-specific config files to specify which one is desired at a given point. Granted, I only know how to do this in VSC, but I'm sure IJ and AS have the feature as well.